File: src/weaponinventorymod/gui/WimGuiNoopCoreInteractionListener.java
package weaponinventorymod.gui;

import com.fs.starfarer.api.campaign.CoreInteractionListener;

final class WimGuiNoopCoreInteractionListener implements CoreInteractionListener {
    static final WimGuiNoopCoreInteractionListener INSTANCE = new WimGuiNoopCoreInteractionListener();

    private WimGuiNoopCoreInteractionListener() {
    }

    @Override
    public void coreUIDismissed() {
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiNoopCoreInteractionListener.kt
package weaponsprocurement.gui

import com.fs.starfarer.api.campaign.CoreInteractionListener

class WimGuiNoopCoreInteractionListener private constructor() : CoreInteractionListener {
    override fun coreUIDismissed() {
    }

    companion object {
        @JvmField
        val INSTANCE: WimGuiNoopCoreInteractionListener = WimGuiNoopCoreInteractionListener()
    }
}
